sys/net/gnrc/network_layer/ipv6/nib/nib_nc.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/netif.h"
+
+#include "net/gnrc/ipv6/nib/nc.h"
+
+#include "_nib-internal.h"
+
+int gnrc_ipv6_nib_nc_set(const ipv6_addr_t *ipv6, unsigned iface,
+                         const uint8_t *l2addr, size_t l2addr_len)
+{
+    _nib_onl_entry_t *node;
+
+    assert((ipv6 != NULL) && (l2addr != NULL));
+    assert(l2addr_len <= GNRC_IPV6_NIB_L2ADDR_MAX_LEN);
+    assert(iface <= KERNEL_PID_LAST);
+    mutex_lock(&_nib_mutex);
+    node = _nib_nc_add(ipv6, iface, GNRC_IPV6_NIB_NC_INFO_NUD_STATE_UNMANAGED);
+    if (node == NULL) {
+        mutex_unlock(&_nib_mutex);
+        return -ENOMEM;
+    }
+#if GNRC_IPV6_NIB_CONF_ARSM
+    memcpy(node->l2addr, l2addr, l2addr_len);
+    node->l2addr_len = l2addr_len;
+#else
+    (void)l2addr;
+    (void)l2addr_len;
+#endif
+    node->info &= ~(GNRC_IPV6_NIB_NC_INFO_AR_STATE_MASK |
+                    GNRC_IPV6_NIB_NC_INFO_NUD_STATE_MASK);
+    node->info |= (GNRC_IPV6_NIB_NC_INFO_AR_STATE_MANUAL |
+                   GNRC_IPV6_NIB_NC_INFO_NUD_STATE_UNMANAGED);
+    mutex_unlock(&_nib_mutex);
+    return 0;
+}
+
+void gnrc_ipv6_nib_nc_del(const ipv6_addr_t *ipv6)
+{
+    _nib_onl_entry_t *node = NULL;
+
+    mutex_lock(&_nib_mutex);
+    while ((node = _nib_onl_iter(node)) != NULL) {
+        if (ipv6_addr_equal(ipv6, &node->ipv6)) {
+            _nib_nc_remove(node);
+            break;
+        }
+    }
+    mutex_unlock(&_nib_mutex);
+}
+
+void gnrc_ipv6_nib_nc_mark_reachable(const ipv6_addr_t *ipv6)
+{
+    _nib_onl_entry_t *node = NULL;
+
+    mutex_lock(&_nib_mutex);
+    while ((node = _nib_onl_iter(node)) != NULL) {
+        if ((node->mode & _NC) && ipv6_addr_equal(ipv6, &node->ipv6)) {
+            /* only set reachable if not unmanaged */
+            if ((node->info & GNRC_IPV6_NIB_NC_INFO_NUD_STATE_MASK)) {
+                _nib_nc_set_reachable(node);
+            }
+            break;
+        }
+    }
+    mutex_unlock(&_nib_mutex);
+}
+
+bool gnrc_ipv6_nib_nc_iter(unsigned iface, void **state,
+                           gnrc_ipv6_nib_nc_t *entry)
+{
+    _nib_onl_entry_t *node = *state;
+
+    mutex_lock(&_nib_mutex);
+    while ((node = _nib_onl_iter(node)) != NULL) {
+        if ((node->mode & _NC) &&
+            ((iface == 0) || (_nib_onl_get_if(node) == iface))) {
+            _nib_nc_get(node, entry);
+            *state = node;
+            mutex_unlock(&_nib_mutex);
+            return true;
+        }
+    }
+    *state = NULL;
+    mutex_unlock(&_nib_mutex);
+    return false;
+}
+
+static const char *_nud_str[] = {
```

Comment:
> I am reading this a lot lately on github. I don't think that referring to other places in the current code base is justification enough, especially when there's no documented coding convention for such cases. I am aware of those other places, but subject of my review is this PR. To be honest, these comments demotivate me as a reviewer.

In this case it wasn't meant as a justification for bad style, but as "others do it too, so I assume LTO works here" (which I then went on to verify ;-))